topology/probes/blockdev/blockdev.go

```diff
@@ -0,0 +1,515 @@
+	DiscAln      json.RawMessage `json:"disc-aln"`  // int64
+	DiscGran     string          `json:"disc-gran"`
+	DiscMax      string          `json:"disc-max"`
+	DiscZero     json.RawMessage `json:"disc-zero"` // bool
+	Fsavail      string          `json:"fsavail"`
+	Fssize       string          `json:"fssize"`
+	Fstype       string          `json:"fstype"`
+	FsusePercent string          `json:"fsuse%"`
+	Fsused       string          `json:"fsused"`
+	Group        string          `json:"group"`
+	Hctl         string          `json:"hctl"`
+	Hotplug      json.RawMessage `json:"hotplug"` // bool
+	Kname        string          `json:"kname"`
+	Label        string          `json:"label"`
+	LogSec       json.RawMessage `json:"log-sec"` // int64
+	MajMin       string          `json:"maj:min"`
+	MinIo        json.RawMessage `json:"min-io"` // int64
+	Mode         string          `json:"mode"`
+	Model        string          `json:"model"`
+	Mountpoint   string          `json:"mountpoint"`
+	Name         string          `json:"name"`
+	OptIo        json.RawMessage `json:"opt-io"` // int64
+	Owner        string          `json:"owner"`
+	Partflags    string          `json:"partflags"`
+	Partlabel    string          `json:"partlabel"`
+	Parttype     string          `json:"parttype"`
+	Partuuid     string          `json:"partuuid"`
+	Path         string          `json:"path"`
+	PhySec       json.RawMessage `json:"hpy-sec"` // int64
+	Pkname       string          `json:"pkname"`
+	Pttype       string          `json:"pttype"`
+	Ptuuid       string          `json:"ptuuid"`
+	Ra           json.RawMessage `json:"ra"`   // int64
+	Rand         json.RawMessage `json:"rand"` // bool
+	Rev          string          `json:"rev"`
+	Rm           json.RawMessage `json:"rm"`      // bool
+	Ro           json.RawMessage `json:"ro"`      // bool
+	Rota         json.RawMessage `json:"rota"`    // bool
+	RqSize       json.RawMessage `json:"rq-size"` // int64
+	Sched        string          `json:"sched"`
+	Serial       string          `json:"serial"`
+	Size         string          `json:"size"`
+	State        string          `json:"state"`
+	Subsystems   string          `json:"subsystems"`
+	Tran         string          `json:"tran"`
+	Type         string          `json:"type"`
+	UUID         string          `json:"uuid"`
+	Vendor       string          `json:"vendor"`
+	Wsame        string          `json:"wsame"`
+	WWN          string          `json:"wwn"`
+}
+
+// Devices used for JSON parsing
+type Devices struct {
+	Blockdevices []BlockDevice `json:"blockdevices"`
+}
+
+type blockdevInfo struct {
+	ID   string
+	Node *graph.Node
+}
+
+// ProbeHandler describes a block device graph that enhances the graph
+type ProbeHandler struct {
+	common.RWMutex
+	Ctx         tp.Context
+	blockdevMap map[string]blockdevInfo
+	wg          sync.WaitGroup
+	Groups      map[string]*graph.Node
+}
+
+func (p *ProbeHandler) getPath(blockdev BlockDevice) string {
```

Comment:
nit: what about making `getPath` a method on BlockDevice instead of ProbeHandler